src/Data/TreeDiff/Parser.hs

```diff
@@ -111,3 +120,44 @@ valid c = isAlphaNum c || isSymbol c || isPunctuation c
 
 valid' :: Char -> Bool
 valid' c = valid c && c `notElem` "[](){}`\","
+
+{-| Parse an `Expr` from a type's `Show` instance.  This can come in handy if a
+    type already has a `Show` instance and you don't want to have to derive
+    `ToExpr` for that type and all of its dependencies.
+-}
+showToExpr :: Show a => a -> Either ShowParseFailed Expr
+showToExpr a =
+    case Parsec.parse exprParser "" (show a) of
```

Comment:
This doesn't work. `exprParser` is not parser for Haskell expressions. Close syntax, but not the right one.